Extensions/xkit_patches.js

```diff
@@ -200,6 +246,11 @@ XKit.extensions.xkit_patches = new Object({
 							response: {
 								status: xhr.status,
 								responseText: xhr.response,
+								responseObj: (() => {
+									try {
+										return JSON.parse(xhr.response);
+									} catch (e) { return null; }
+								})(),
```

Comment:
the iife here is kind of weird, and failing silently might be confusing later. Maybe adding this as a `json()` function or getter when it gets back into the XKit namespace would make more sense?

Reply:
i don't know how to do this; we can't define an actual function here, nor can one be defined on the response object in the handler function

Reply:
Why can't we define one in the handler object?

Reply:
`Error: "Not allowed to define cross-origin object as property on [Object] or [Array] XrayWrapper"`

Reply:
Oh, weird, works fine in chrome. Let's just json.parse(json.stringify()) the response before doing anything with it.

Reply:
if we're that desperate i guess it should work!